aslam_cv_pipeline/src/visual-npipeline.cc

```diff
@@ -48,6 +48,7 @@ VisualNPipeline::~VisualNPipeline() {
 }
 
 void VisualNPipeline::shutdown() {
+  std::unique_lock<std::mutex> lock(mutex_);
```

Comment:
is this required? could the shutdown_ be an atomic<bool>?

Reply:
`shutdown_` is atomic. I think we could savely move the lock in front of` thread_pool_->stop();`

Reply:
Actually, I think we can remove the lock entirely. @mbuerki What was the reason for adding this?

Reply:
But if we remove it, what happens if another thread is still executing a function of this class while the destructor is called?

Reply:
Sorry for monologue: In case of proper ownership handling the last case should never occur...